src/main/java/org/takes/rs/RsWithType.java

```diff
@@ -46,25 +47,67 @@
     private static final String HEADER = "Content-Type";
 
     /**
-     * Ctor.
+     * The name of the parameter allowing to define the character set.
+     */
+    private static final String CHARSET = "charset";
+
+    /**
+     * Constructs a {@code RsWithType} that will add the content type header to
+     * the response using the specified type as media type.
+     * <p>The resulting header is of type <code>Content-Type: media-type</code>.
      * @param res Original response
      * @param type Content type
      */
     public RsWithType(final Response res, final CharSequence type) {
-        super(RsWithType.make(res, type));
+        this(res, type, null);
+    }
+
+    /**
+     * Constructs a {@code RsWithType} that will add the content type header to
+     * the response using the specified type as media type and the specified
+     * charset as charset parameter value.
+     * <p>The resulting header
+     * is of type <code>Content-Type: media-type; charset=charset-value</code>.
+     * @param res Original response
+     * @param type Content type
+     * @param charset The character set to add in the content type header
+     */
+    public RsWithType(final Response res, final CharSequence type,
+        final Charset charset) {
+        super(RsWithType.make(res, type, charset));
     }
 
     /**
-     * Ctor.
+     * Factory allowing to create {@code Response} with the corresponding
+     * content type and character set.
      * @param res Original response
      * @param type Content type
+     * @param charset The character set to add to the content type header. If
+     *  {@code null} no character set will be added to the content type header
      * @return Response
      */
-    private static Response make(final Response res, final CharSequence type) {
-        return new RsWithHeader(
-            new RsWithoutHeader(res, RsWithType.HEADER),
-            RsWithType.HEADER, type
-        );
+    private static Response make(final Response res, final CharSequence type,
+        final Charset charset) {
+        final Response response;
+        if (charset == null) {
```

Comment:
@essobedo `null` in Takes is simply prohibited, please see README. You can try using `Optional` or custom object instead.